app/scripts/modules/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -333,13 +341,45 @@ angular
             command.viewState.imageId = serverGroup.launchConfig.imageId;
           }
 
+          if (serverGroup.launchTemplate) {
+            const { launchTemplateData } = serverGroup.launchTemplate;
+            const maxPrice =
+              launchTemplateData.metadataOptions &&
+              launchTemplateData.metadataOptions.spotOptions &&
+              launchTemplateData.metadataOptions.spotOptions.maxPrice;
+            const ipv6Addresses = _.flatMap(launchTemplateData.networkInterfaces, i => i.ipv6Addresses) || [];
+
+            angular.extend(command, {
+              instanceType: launchTemplateData.instanceType,
+              iamRole: launchTemplateData.iamInstanceProfile.name,
+              keyPair: launchTemplateData.keyName,
+              associatePublicIpAddress: ipv6Addresses.length > 0,
+              ramdiskId: launchTemplateData.ramdiskId,
+              instanceMonitoring: launchTemplateData.monitoring.enabled,
+              ebsOptimized: launchTemplateData.ebsOptimized,
+              spotPrice: maxPrice || undefined,
+              requireIMDSv2:
+                launchTemplateData.metadataOptions && launchTemplateData.metadataOptions.httpTokens === 'required',
+            });
+
+            if (launchTemplateData.userData) {
+              command.base64UserData = launchTemplateData.userData;
+            }
+            command.viewState.imageId = launchTemplateData.imageId;
+          }
+
           if (mode === 'clone' && serverGroup.image && serverGroup.image.name) {
             command.amiName = serverGroup.image.name;
           }
 
           if (serverGroup.launchConfig && serverGroup.launchConfig.securityGroups.length) {
             command.securityGroups = serverGroup.launchConfig.securityGroups;
           }
+
+          if (serverGroup.launchTemplate && serverGroup.launchTemplate.launchTemplateData.securityGroups.length) {
```

Comment:
The backend is setting security groups on network interfaces. (Ran into issues with cloneServerGroups when the groups aren’t set at the network interface level.

Reply:
Ok, is it problematic to set them here too or should I just set them on both `securityGroups` and `networkInterfaces`?

Reply:
Just securityGroups should be fine. Groups can be found here `serverGroup.launchTemplate.launchTemplateData.networkInterfaces*.groups`.  Before you change anything, let me try a couple of things.